### ISSUE TITLE
Limit the number of code hosts/remote repo connections

### ISSUE THREAD

**Issue by taylorsperry** [sourcegraph/sourcegraph#48785]

**Comment by vovakulikov** [comment]
@taylorsperry just to make sure we agree on this. This limitation will be "hardcoded" this means that there won't be any special license/tag/plan for unlimited Sourcegraph App, right?

**Comment by taylorsperry** [comment]
Yes, I think hardcoded is fine for now. Eventually the Superdev license will unlock unlimited codehosts/repos, but that's a project for post-Starship.

**Comment by keegancsmith** [comment]
@vovakulikov @gl-srgr this issue seems independent of setup wizard. However, I am assuming you need to be aware of this in the wizard to give a good user experience. Any tips on what APIs we should provide / good errors / etc? Any tips on APIs you have added / use that should be made aware of these limits?

**Comment by vovakulikov** [comment]
@keegancsmith since this limitation is hardcoded and won't change over time during an instance run I think we will use just the isSourcegraph boolean flag, which is already in the JSContext, and add some validation to the setup form (disable buttons if we have more than 1 remote code host) 

I think from an API point of view, we need to 
- gate creation of external service if there is already more than just one connected (Sourcegraph app-specific error in return on this mutation?)
- Repositories limit seems a bit complex, we can't have anything about this limitation on the client since we don't know anything about the repositories number
- Maybe we could add some specific to this case (when we limit the number of repositories to the external sync syncJobs field that we stopped syncing since we've resolved more than 10 repositories already)

**Comment by philipp-spiess** [comment]
@vovakulikov I think your suggestion for the frontend part sounds good. I'll make sure this is reflected in the admin backend.

I’m thinking of doing something like this:

<img width="995" alt="Screenshot 2023-03-08 at 14 44 27" src="https://user-images.githubusercontent.com/458591/223728705-0ce19fb6-0b47-439b-8a8f-fa7efe3d1c3e.png">

What do you all think?

**Comment by philipp-spiess** [comment]
Maybe the text should be `(2 of 10 repositories)` hehe

**Comment by philipp-spiess** [comment]
<img width="983" alt="Screenshot 2023-03-08 at 14 45 42" src="https://user-images.githubusercontent.com/458591/223728958-718e88b6-b69f-4583-8416-bb68649227ca.png">

And maybe in here we also show the blue box and mention that only 10 repositories will be synced from the config?

**Comment by philipp-spiess** [comment]
And for the code host edit/view pages we add a simple hint as well (one uses the blue box and one adds the limitations to the total count)

<img width="949" alt="Screenshot 2023-03-08 at 14 49 14" src="https://user-images.githubusercontent.com/458591/223730324-0f94746e-818a-47f9-b868-6924b9e01a0b.png">
<img width="993" alt="Screenshot 2023-03-08 at 14 48 18" src="https://user-images.githubusercontent.com/458591/223730329-26204208-69b8-4f9b-9b44-0dcf5cbcfcd4.png">


cc @almeidapaulooliveira if you have some other quick ideas on how to update the UI to make it clear which limits are in place and why.

**Comment by almeidapaulooliveira** [comment]
Good solutions, @philipp-spiess. I agree with all of them.

**Comment by vovakulikov** [comment]
Cool, @philipp-spiess, I will add the same limitation UI that you proposed above to the wizard remote code step 😉

**Comment by vovakulikov** [comment]
@philipp-spiess @keegancsmith can we close this one? I've seen that alerts UI and client checks have been merged but not sure about API gating

**Comment by philipp-spiess** [comment]
@vovakulikov The FE changes for this are done but I’m not sure about the BE cc @keegancsmith

**Comment by slimsag** [comment]
Probably very low priority until we hear user feedback about app being slow, which we believe to be from them adding too much code.
